feature/security/gnsi/acctz/tests/record_subscribe_partial/record_subscribe_partial_test.go

```diff
@@ -0,0 +1,284 @@
+			r <- recordRequestResult{
+				record: response,
+				err:    err,
+			}
+		}(r)
+
+		var done bool
+		var resp recordRequestResult
+
+		select {
+		case rr := <-r:
+			resp = rr
+		case <-time.After(10 * time.Second):
+			done = true
+		}
+
+		if done {
+			t.Log("Done receiving records...")
+			break
+		}
+
+		if resp.err != nil {
+			t.Fatalf("Failed receiving record response, error: %s", resp.err)
+		}
+
+		if resp.record.GetHistoryIstruncated() {
+			t.Fatalf("History is truncated but it shouldn't be, Record Details: %s", prettyPrint(resp.record))
+		}
+
+		if !resp.record.Timestamp.AsTime().After(startTime) {
+			// Skipping record, was before test start time.
+			continue
+		}
+
+		// Check that the timestamp for the record is between our start/stop times for our rpc.
+		timestamp := resp.record.Timestamp.AsTime()
+		if timestamp.UnixMilli() == lastTimestampUnixMillis {
+			// This ensures that timestamps are actually changing for each record.
+			t.Fatalf("Timestamp is the same as the previous timestamp, this shouldn't be possible!, Record Details: %s", prettyPrint(resp.record))
+		}
+		lastTimestampUnixMillis = timestamp.UnixMilli()
+
+		// -2 for a little breathing room since things may not be perfectly synced up time-wise.
+		if records[recordIdx].StartTime.Unix()-2 > timestamp.Unix() {
+			t.Fatalf(
+				"Record timestamp is prior to rpc start time timestamp, rpc start timestamp %d, record timestamp %d, Record Details: %s",
+				records[recordIdx].StartTime.Unix()-2,
+				timestamp.Unix(),
+				prettyPrint(resp.record),
+			)
+		}
+
+		// Done time (that we recorded when making the rpc) + 2 second for some breathing room.
+		if records[recordIdx].DoneTime.Unix()+2 < timestamp.Unix() {
+			t.Fatalf(
+				"Record timestamp is after rpc end timestamp, rpc end timestamp %d, record timestamp %d, Record Details: %s",
+				records[recordIdx].DoneTime.Unix()+2,
+				timestamp.Unix(),
+				prettyPrint(resp.record),
+			)
+		}
+
+		serviceType := resp.record.GetGrpcService().GetServiceType()
+		if records[recordIdx].RPCType != serviceType {
+			t.Fatalf("Service type not correct, got %q, want %q, Record Details: %s", serviceType, records[recordIdx].RPCType, prettyPrint(resp.record))
+		}
+
+		servicePath := resp.record.GetGrpcService().GetRpcName()
+		if records[recordIdx].RPCPath != servicePath {
+			t.Fatalf("Service path not correct, got %q, want %q, Record Details: %s", servicePath, records[recordIdx].RPCPath, prettyPrint(resp.record))
+		}
+
+		if records[recordIdx].RPCPayload != "" {
+			// It seems like it *could* truncate payloads so that may come up at some point
+			// which would obviously make this comparison not work, but for the simple RPCs in
+			// this test that probably shouldn't be happening.
+			gotServicePayload := resp.record.GetGrpcService().GetProtoVal().String()
+			wantServicePayload := records[recordIdx].RPCPayload
+			if !strings.EqualFold(gotServicePayload, wantServicePayload) {
```

Comment:
Is this preferred over cmp.Equal? or cmp.Diff() and using the diff output to put in the error message?

```golang
if diff := cmp.Diff(got, want); diff != "" {
   t.Errorf("got diff in got/want: %s", diff)
}
```